packages/design-system/src/components/InlineEditing/Primitives/InlineEditingPrimitive.tsx

```diff
@@ -38,16 +36,40 @@ export type OnEditEvent =
 
 export type InlineEditingPrimitiveProps = {
 	loading?: boolean;
-	onEdit?: (event: OnEditEvent, newValue: string) => void;
 	onCancel?: () => void;
 	onToggle?: (isEditionMode: boolean) => void;
-	defaultValue?: string;
 	label: string;
 	required?: boolean;
 	placeholder: string;
 	ariaLabel?: string;
 	renderValueAs?: ElementType | ReactElement;
 	mode: 'single' | 'multi';
+
+	/**
+	 * (Optional) Initial value displayed in component.
+	 * To set to display initial value while use this component as uncontrolled one.
+	 */
+	defaultValue?: string;
+
+	/**
+	 * (Optional) In uncontrolled way, listen value update.
+	 * @param event change event
+	 * @param newValue new value set in
+	 */
+	onEdit?: (event: OnEditEvent, newValue: string) => void;
+
+	/**
+	 * (Optional) Value displayed in component.
+	 * To set to use component as controlled one.
+	 */
+	value?: string;
+
+	/**
+	 * (Optional) In controlled way, handle value update.
+	 * It must be set to use this component as controlled component.
+	 * @param newValue new value filled
+	 */
+	onChangeValue?: (newValue: string) => void;
```

Comment:
When it comes to define an API for a component that have 2 exclusives mode, i would prefer make it explicit with the typescript interface.

Like this way:
https://github.com/Talend/ui/pull/4720/files#diff-65104284714bc80639dde175162db58c4d8cb948751dd62e72377ce9c6906941R38

Cause if 2 sets of props is only working together, it's easier to understand rather than trying to pick the right ones